setup/install/providers/aws/aws-ec2.md

```diff
@@ -102,47 +113,257 @@ aws cloudformation deploy --stack-name spinnaker-managed-infrastructure-setup --
+
+For the example below, the AWS Account **spinnakerManaging** assumes the **spinnakerManaged** role in the AWS accounts **develop** and **staging**. The account **spinnakerManaging** is where Spinnaker lives.
+
+A great use case for this set up is to deploy pre-built AWS AMIs to AWS EC2.
+
+![Example AWS IAM structure for Spinnaker AWS Provider](/setup/install/providers/images/example-aws-provider.svg)
+
+Before you start, create a table that maps the account names to account IDs for your desired set up. An example table is shown:
+
+| Name              | Account Id   |
+|-------------------|--------------|
+| spinnakerManaging | 100000000001 |
+| develop           | 200000000002 |
+| staging           | 300000000003 |
+
+These examples are used in the subsequent sections.
+
+### AWS IAM user or roles
+
+For IAM, you can either create IAM users or roles based on your requirements:
+
+1. Set Up AWS IAM structure and use an AWS IAM User *spinnakerManaging* with AccessKey and Secret. This option is useful for creating users with AWS Access Key and secret. This is a common configuration.
+2. Using AWS IAM Roles, such as creating a *ManagingRole* and *ManagedRoles*. This option might be needed for environments that have extra security considerations. The EC2 instance where Spinnaker is used to deploy AWS resources has the *ManagingRole* attached. The *ManagingRole* contains the AWS IAM policies required to manage AWS resources.
+
+### Create AWS IAM user
+
+1. Navigate to the AWS IAM console.
+2. Switch to the **spinnakerManaging** AWS Account.
+3. Add a user and name it **spinnakerManaging**.
+4. Check the “Programmatic access” checkbox.
+5. Add tags to help you identify this user, such as "spinnakerManaging".
+6. Create the user.
+7. Save the “Access key ID” and “Secret access key” somewhere secure, such as a secret management system like Vault, AWS Secrets manager, or 1Password.
+8. Save the AWS ARN for the user, for example: `arn:aws:iam::100000000001:user/spinnakerManaging`. You need the ARN later when you configure the managed accounts to trust the managing account (IAM user).
+
+### Create Roles
+
+#### Create Managed Roles in each target AWS Account
+
+First, create the **spinnakerManaged** role for the **develop ID=200000000002**. Then, repeat the same steps to create **spinnakerManaged** in **staging ID=300000000003**.
+
+1. Navigate to the AWS IAM console.
+2. Switch to the AWS Account you want to create the roll for.
+3. Go to **Roles > Create Role**.
+4. Select EC2. You can change this later, because we want to specify an explicit consumer of this role in a later stage.
+5. For permissions, search for "PowerUserAccess" and select this policy. This gives the role permission to access AWS services.
```

Comment:
PowerUserAccess is broader access than Spinnaker really needs; there used to be a very specific policy document doc that had what Spinnaker needs; we should refer to that here. I think that Armory at least has an up-to-date, narrower policy to refer to.

Reply:
what access do you recommend? 
the policy is AWS managed it should be ok...

Reply:
The PowerUserAccess policy is extremely broad. Unfortunately it looks like the Armory docs also now recommend PowerUserAccess.

This is the policy I'm using now, which at least limits access to the core services Spinnaker needs, although it would need to be broadened to account for CloudFormation and Lambda and potentially other small revisions. This or a similar policy used to be present in the Spinnaker docs (or perhaps the Armory docs).
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "Stmt1486065689000",
            "Effect": "Allow",
            "Action": [
                "autoscaling:CreateAutoScalingGroup",
                "autoscaling:CreateLaunchConfiguration",
                "autoscaling:CreateOrUpdateTags",
                "autoscaling:DeleteAutoScalingGroup",
                "autoscaling:DeleteLaunchConfiguration",
                "autoscaling:DeletePolicy",
                "autoscaling:DeleteScheduledAction",
                "autoscaling:DeleteTags",
                "autoscaling:DescribeAutoScalingGroups",
                "autoscaling:DescribeLaunchConfigurations",
                "autoscaling:DescribeLoadBalancers",
                "autoscaling:DescribePolicies",
                "autoscaling:DescribeScalingActivities",
                "autoscaling:DescribeScheduledActions",
                "autoscaling:DetachInstances",
                "autoscaling:DisableMetricsCollection",
                "autoscaling:EnableMetricsCollection",
                "autoscaling:PutLifecycleHook",
                "autoscaling:PutNotificationConfiguration",
                "autoscaling:PutScalingPolicy",
                "autoscaling:PutScheduledUpdateGroupAction",
                "autoscaling:ResumeProcesses",
                "autoscaling:SuspendProcesses",
                "autoscaling:TerminateInstanceInAutoScalingGroup",
                "autoscaling:UpdateAutoScalingGroup",
                "cloudwatch:DeleteAlarms",
                "cloudwatch:DescribeAlarms",
                "cloudwatch:GetMetricStatistics",
                "cloudwatch:ListMetrics",
                "cloudwatch:PutMetricAlarm",
                "ec2:AttachClassicLinkVpc",
                "ec2:AuthorizeSecurityGroupIngress",
                "ec2:CreateNetworkInterface",
                "ec2:CreateSecurityGroup",
                "ec2:CreateTags",
                "ec2:DeleteSecurityGroup",
                "ec2:DeleteTags",
                "ec2:DescribeAccountAttributes",
                "ec2:DescribeAddresses",
                "ec2:DescribeAvailabilityZones",
                "ec2:DescribeClassicLinkInstances",
                "ec2:DescribeImages",
                "ec2:DescribeInstances",
                "ec2:DescribeKeyPairs",
                "ec2:DescribeRegions",
                "ec2:DescribeReservedInstances",
                "ec2:DescribeReservedInstancesOfferings",
                "ec2:DescribeSecurityGroups",
                "ec2:DescribeSpotPriceHistory",
                "ec2:DescribeSubnets",
                "ec2:DescribeTags",
                "ec2:DescribeVpcClassicLink",
                "ec2:DescribeVpcs",
                "ec2:GetConsoleOutput",
                "ec2:ModifyImageAttribute",
                "ec2:ModifyInstanceAttribute",
                "ec2:RebootInstances",
                "ec2:RevokeSecurityGroupIngress",
                "ec2:TerminateInstances",
                "elasticloadbalancing:ApplySecurityGroupsToLoadBalancer",
                "elasticloadbalancing:ConfigureHealthCheck",
                "elasticloadbalancing:CreateListener",
                "elasticloadbalancing:CreateLoadBalancer",
                "elasticloadbalancing:CreateLoadBalancerListeners",
                "elasticloadbalancing:CreateLoadBalancerPolicy",
                "elasticloadbalancing:CreateRule",
                "elasticloadbalancing:CreateTargetGroup",
                "elasticloadbalancing:DeleteListener",
                "elasticloadbalancing:DeleteLoadBalancer",
                "elasticloadbalancing:DeleteLoadBalancerListeners",
                "elasticloadbalancing:DeleteRule",
                "elasticloadbalancing:DeleteTargetGroup",
                "elasticloadbalancing:DeregisterInstancesFromLoadBalancer",
                "elasticloadbalancing:DeregisterTargets",
                "elasticloadbalancing:DescribeInstanceHealth",
                "elasticloadbalancing:DescribeListeners",
                "elasticloadbalancing:DescribeLoadBalancerAttributes",
                "elasticloadbalancing:DescribeLoadBalancerPolicies",
                "elasticloadbalancing:DescribeLoadBalancers",
                "elasticloadbalancing:DescribeRules",
                "elasticloadbalancing:DescribeTargetGroupAttributes",
                "elasticloadbalancing:DescribeTargetGroups",
                "elasticloadbalancing:DescribeTargetHealth",
                "elasticloadbalancing:ModifyListener",
                "elasticloadbalancing:ModifyLoadBalancerAttributes",
                "elasticloadbalancing:ModifyTargetGroup",
                "elasticloadbalancing:ModifyTargetGroupAttributes",
                "elasticloadbalancing:RegisterInstancesWithLoadBalancer",
                "elasticloadbalancing:RegisterTargets",
                "elasticloadbalancing:SetLoadBalancerPoliciesOfListener",
                "elasticloadbalancing:SetSecurityGroups",
                "iam:ListServerCertificates",
                "iam:PassRole"
            ],
            "Resource": [
                "*"
            ]
        }
    ]
}```

Reply:
A customer managed policy is cumbersome for an organization. Is an interesting concept to try to reduce risk but is just too much to handle.

Reply:
With cloudformation stacksets and AWS organizations it is quite feasible to have a customer managed policy.

The power user policy is an irresponsible one and it reflects poorly on the product to recommend it.

Reply:
I understand. If thats the case i highly encourage you to improve the Cloud formation templates that use the AWS IAM policies at https://github.com/spinnaker/spinnaker.github.io/tree/master/downloads/aws